Mage.Sets/src/mage/cards/g/GiadaFontOfHope.java

```diff
@@ -0,0 +1,130 @@
+import mage.abilities.keyword.VigilanceAbility;
+import mage.abilities.mana.ConditionalColoredManaAbility;
+import mage.abilities.mana.conditional.ConditionalSpellManaBuilder;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.events.EntersTheBattlefieldEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+/**
+ * @author woshikie
+ */
+public final class GiadaFontOfHope extends CardImpl {
+
+    private static final FilterPermanent OTHER_ANGEL_YOU_CONTROL_FILTER = new FilterControlledCreaturePermanent(SubType.ANGEL, "other Angel you control");
+    private static final FilterSpell ANGEL_SPELL_FILTER = new FilterSpell("an Angel spell");
+
+    static {
+        OTHER_ANGEL_YOU_CONTROL_FILTER.add(AnotherPredicate.instance);
+        ANGEL_SPELL_FILTER.add(SubType.ANGEL.getPredicate());
+    }
+
+    public GiadaFontOfHope(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{W}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.ANGEL);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Each other Angel you control enters the battlefield with an additional +1/+1 counter on it for each Angel you already control.
+        this.addAbility(
+                new SimpleStaticAbility(
+                        Zone.BATTLEFIELD,
+                        new GiadaFontOfHopeEntersBattlefieldEffect()
+                )
+        );
+
+        // {T}: Add {W}. Spend this mana only to cast an Angel spell.
+        this.addAbility(
+                new ConditionalColoredManaAbility(
+                        Mana.WhiteMana(1),
+                        new ConditionalSpellManaBuilder(ANGEL_SPELL_FILTER)
+                )
+
+        );
+    }
+
+    private GiadaFontOfHope(final GiadaFontOfHope card) {
+        super(card);
+    }
+
+    @Override
+    public GiadaFontOfHope copy() {
+        return new GiadaFontOfHope(this);
+    }
+}
+
+
+class GiadaFontOfHopeEntersBattlefieldEffect extends ReplacementEffectImpl {
+
+    public GiadaFontOfHopeEntersBattlefieldEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.BoostCreature);
+        staticText = "Each other Angel you control enters the battlefield with an additional +1/+1 counter on it for each Angel you already control.";
+    }
+
+    public GiadaFontOfHopeEntersBattlefieldEffect(GiadaFontOfHopeEntersBattlefieldEffect effect) {
```

Comment:
This should be `private GiadaFontOfHopeEntersBattlefieldEffect(final GiadaFontOfHopeEntersBattlefieldEffect effect)`

Reply:
cleaned up in ac18315c39815165f80d74d3d16bc348edffa099